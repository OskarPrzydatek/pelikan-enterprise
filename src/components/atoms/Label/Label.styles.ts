import styled from 'styled-components';

export const Label = styled.label`
  font-weight: 600;
  color: ${({ theme }) => theme.colors.black};
  font-family: ${({ theme }) => theme.fonts.main};
  font-size: ${({ theme }) => theme.fontSizes.xxs};

  @media ${({ theme }) => theme.breakpoints.phone} {
    font-size: ${({ theme }) => theme.fontSizes.xxs};
  }

  @media ${({ theme }) => theme.breakpoints.tablet} {
    font-size: ${({ theme }) => theme.fontSizes.xs};
  }

  @media ${({ theme }) => theme.breakpoints.desktop} {
    font-size: ${({ theme }) => theme.fontSizes.xs};
  }
`;
