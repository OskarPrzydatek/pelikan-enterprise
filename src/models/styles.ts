import {
  DefaultTheme,
  FlattenInterpolation,
  ThemeProps,
} from 'styled-components';

export interface ICSS {
  css?: FlattenInterpolation<ThemeProps<DefaultTheme>>;
}
