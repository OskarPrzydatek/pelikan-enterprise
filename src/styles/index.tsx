export * from './layouts';
export { theme } from './theme';
