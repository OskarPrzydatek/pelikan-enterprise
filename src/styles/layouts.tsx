import styled, { createGlobalStyle } from 'styled-components';

export const GlobalStyleLayout = createGlobalStyle`
  * {
		box-sizing: border-box;
    outline: none;
	} 

  body {
    margin: 0;
    padding: 0;
  }

  #root {
    height: 100vh;
    display: grid;
    grid-template-rows: .5fr 8fr .5fr;
  }

	a {
		color: inherit;
		text-decoration: none;
	}

	ul {
		padding: 0;
    margin: 0;
		list-style-type: none;
	}
`;

export const MainLayout = styled.main`
  padding: ${({ theme }) => theme.spaces.l};
`;

export const PageLayout = styled.div`
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;
