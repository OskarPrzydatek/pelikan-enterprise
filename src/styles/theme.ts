import { DefaultTheme } from 'styled-components';

export const theme: DefaultTheme = {
  colors: {
    red: '#F23557',
    lightBlue: '#98BFF2',
    darkBlue: '#013059',
    gray: '#5F5F5F',
    black: '#2F2F2F',
    opacityBlack: '#2F2F2FCC',
    shadowBlack: '#00000040',
    white: '#FFFFFF',
  },
  breakpoints: {
    phone: '(min-width: 420px)',
    tablet: '(min-width: 920px)',
    desktop: '(min-width: 1200px)',
  },
  fonts: {
    main: `'Inter', sans-serif`,
  },
  fontSizes: {
    xxs: '.2rem',
    xs: '.5rem',
    s: '1rem',
    m: '1.2rem',
    l: '1.5rem',
    xl: '1.8rem',
    xxl: '2rem',
  },
  spaces: {
    xxs: '.2rem',
    xs: '.5rem',
    s: '.8rem',
    m: '1.2rem',
    l: '1.5rem',
    xl: '1.8rem',
    xxl: '2rem',
  },
};
