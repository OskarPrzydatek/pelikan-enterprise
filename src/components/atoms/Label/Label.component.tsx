import {
  DefaultTheme,
  FlattenInterpolation,
  ThemeProps,
} from 'styled-components';

import * as S from './Label.styles';

interface ILabel {
  htmlFor: string;
  children: string;
  css?: FlattenInterpolation<ThemeProps<DefaultTheme>>;
  dataTestID?: string;
}

export const Label: React.FC<ILabel> = ({
  htmlFor,
  children,
  css,
  dataTestID,
}: ILabel) => {
  return (
    <S.Label css={css} data-testid={dataTestID} htmlFor={htmlFor}>
      {children}
    </S.Label>
  );
};
