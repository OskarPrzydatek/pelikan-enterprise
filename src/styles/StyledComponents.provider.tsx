import { ThemeProvider } from 'styled-components';

import { GlobalStyleLayout } from './layouts';
import { theme } from './theme';

interface IStyledComponentsProvider {
  children: React.ReactNode;
}

export const StyledComponentsProvider: React.FC<IStyledComponentsProvider> = ({
  children,
}: IStyledComponentsProvider) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyleLayout />
      {children}
    </ThemeProvider>
  );
};
