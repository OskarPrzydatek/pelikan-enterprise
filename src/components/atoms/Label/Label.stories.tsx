import { Meta, StoryObj } from '@storybook/react';

import { StyledComponentsProvider } from '~/providers';

import { Label } from './Label';

const meta: Meta<typeof Label> = {
  title: 'Atoms/Label',
  component: Label,
};

export default meta;
type Story = StoryObj<typeof Label>;

export const LabelStory: Story = {
  render: () => (
    <StyledComponentsProvider>
      <Label htmlFor="label">Label</Label>
    </StyledComponentsProvider>
  ),
};
