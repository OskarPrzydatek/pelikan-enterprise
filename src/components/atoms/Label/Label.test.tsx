import { render, screen } from '@testing-library/react';
import { describe, expect } from 'vitest';

import { StyledComponentsProvider } from '~/styles';

import { Label } from './Label.component';

describe('Label', () => {
  test('component snapshot', () => {
    const view = render(
      <StyledComponentsProvider>
        <Label dataTestID="label-atom" htmlFor="label-atom">
          Label
        </Label>
      </StyledComponentsProvider>
    );
    expect(view).toMatchSnapshot();
  });

  test('ensure label value renders correctly', () => {
    render(
      <StyledComponentsProvider>
        <Label dataTestID="label-atom" htmlFor="label-atom">
          Label
        </Label>
      </StyledComponentsProvider>
    );
    const label = screen.getByTestId('label-atom');
    expect(label).toHaveTextContent('Label');
  });
});
