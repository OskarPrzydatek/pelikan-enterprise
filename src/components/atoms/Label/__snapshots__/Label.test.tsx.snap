// Vitest Snapshot v1

exports[`Label > component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  font-weight: 600;
  color: #21313F;
  font-family: 'Nanum Gothic',sans-serif;
  font-size: .6rem;
}

@media (min-width:390px) {
  .c0 {
    font-size: .6rem;
  }
}

@media (min-width:834px) {
  .c0 {
    font-size: .8rem;
  }
}

@media (min-width:1728px) {
  .c0 {
    font-size: .8rem;
  }
}

<body>
    <div>
      <label
        class="c0"
        data-testid="label-atom"
        for="label-atom"
      >
        Label
      </label>
    </div>
  </body>,
  "container": .c0 {
  font-weight: 600;
  color: #21313F;
  font-family: 'Nanum Gothic',sans-serif;
  font-size: .6rem;
}

@media (min-width:390px) {
  .c0 {
    font-size: .6rem;
  }
}

@media (min-width:834px) {
  .c0 {
    font-size: .8rem;
  }
}

@media (min-width:1728px) {
  .c0 {
    font-size: .8rem;
  }
}

<div>
    <label
      class="c0"
      data-testid="label-atom"
      for="label-atom"
    >
      Label
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
